core/src/main/java/tc/oc/pgm/PGMPlugin.java

```diff
@@ -274,17 +274,40 @@ public void reloadConfig() {
     logger.setLevel(config.getLogLevel());
 
     for (String source : config.getMapSources()) {
-      final MapSourceFactory factory;
+      MapSourceFactory factory = null;
       try {
-        factory =
-            source.equalsIgnoreCase("default")
-                ? DefaultMapSourceFactory.INSTANCE
-                : new SystemMapSourceFactory(source);
+        if (source.equalsIgnoreCase("default"))
+          factory =
+              new GitMapSourceFactory(
+                  "https://github.com/KingOfSquares/DefaultMaps", logger, false);
+        if (source.startsWith("git;")) {
+          String gitSource = source.substring(4);
+          factory = new GitMapSourceFactory(gitSource, logger, true);
+        }
+        if (new File(source).exists()) factory = new SystemMapSourceFactory(source);
+      } catch (StringIndexOutOfBoundsException e) { // Git source parsing exception
+        factory = null; // Trigger the map source error message
       } catch (Throwable t) {
         t.printStackTrace();
         continue;
       }
 
+      if (factory == null) {
+        logger.log(Level.WARNING, "-----MAP SOURCE ERROR-----");
+        logger.log(Level.WARNING, "Map source " + source + " is not a valid source");
+        logger.log(
+            Level.INFO,
+            "To add a map source folder outside of the server folder you have to specify the full path");
+        logger.log(
+            Level.INFO,
+            "To add a git source you have to format your source like this: git;URL;USERNAME;PASSWORD");
+        logger.log(
+            Level.INFO, "Username and password only required if the git repository is private");
+        logger.log(Level.WARNING, source + " will be ignored until restart");
+        logger.log(Level.WARNING, "");
+        continue;
+      }
+
```

Comment:
Too verbose, don't need it.

Reply:
Can i leave some in? I agree i went overboard with this one, but maybe just a
`Map source someMapSource is invalid and will be ignored until restart`
is fine?